frontend/src/libs/hooks/use-app-form/use-app-form.hook.ts

```diff
@@ -13,35 +15,45 @@ import { type ValidationSchema } from '~/libs/types/types.js';
 
 type Parameters<T extends FieldValues = FieldValues> = {
   defaultValues: DeepPartial<T>;
-  validationSchema: ValidationSchema;
+  validationSchema?: ValidationSchema;
+  mode: Mode;
 };
 
 type ReturnValue<T extends FieldValues = FieldValues> = {
   control: Control<T, null>;
   errors: FieldErrors<T>;
   handleSubmit: UseFormHandleSubmit<T>;
-  handleReset: UseFormReset<T>;
+  reset: UseFormReset<T>;
 };
 
 const useAppForm = <T extends FieldValues = FieldValues>({
   validationSchema,
   defaultValues,
+  mode,
```

Comment:
```suggestion
  mode = 'onSubmit',
```
to avoid adding `onSubmit` to all places

Reply:
i received this error: `No value exists in scope for the shorthand property 'mode'. Either declare one or provide an initializer`, when added  `mode = 'onSubmit',`

```
const { control, errors, handleSubmit } = useAppForm<UserSignInRequestDto>({
    defaultValues: DEFAULT_SIGN_IN_PAYLOAD,
    validationSchema: userSignInValidationSchema,
    mode,
  });
```
```
type Parameters<T extends FieldValues = FieldValues> = {
  defaultValues: DeepPartial<T>;
  validationSchema?: ValidationSchema;
  mode: Mode;
};

type ReturnValue<T extends FieldValues = FieldValues> = {
  control: Control<T, null>;
  errors: FieldErrors<T>;
  handleSubmit: UseFormHandleSubmit<T>;
  reset: UseFormReset<T>;
};

const useAppForm = <T extends FieldValues = FieldValues>({
  validationSchema,
  defaultValues,
  mode = 'onSubmit',
}: Parameters<T>): ReturnValue<T> => {
  let parameters: UseFormProps<T> = {
    mode,
    defaultValues,
  };

  if (validationSchema) {
    parameters = {
      ...parameters,
      resolver: joiResolver(validationSchema),
    };
  }
  const {
    control,
    handleSubmit,
    formState: { errors },
    reset,
  } = useForm<T>(parameters);

  return {
    control,
    errors,
    handleSubmit,
    reset,
  };
};
```

Reply:
hmm okay, let's skip it for now